core/src/main/java/org/infinispan/jmx/AbstractJmxRegistration.java

```diff
@@ -127,11 +128,11 @@ public void stop() {
     */
    private String initDomain(MBeanServer mBeanServer) {
       GlobalJmxStatisticsConfiguration globalJmxConfig = globalConfig.globalJmxStatistics();
-      String jmxDomain = JmxUtil.buildJmxDomain(globalJmxConfig.domain(), mBeanServer, groupName);
-      if (!globalJmxConfig.allowDuplicateDomains() && !jmxDomain.equals(globalJmxConfig.domain())) {
-         throw CONTAINER.jmxMBeanAlreadyRegistered(groupName, globalJmxConfig.domain());
+      if (globalJmxConfig.allowDuplicateDomains()) {
```

Comment:
I see you wanted to optimize something here @danberindei : ). I was puzzled by it also and tentatively removed it at some point only to discover it does have some twisted purpose:). It's only benefit was making the failure predictable (ie the exact place where it fails and the failure message) if the domain was occupied and allowDuplicateDomains== false. Otherwise we get a random InstanceAlreadyExistsException elsewhere, which was not even showing up directly because old JmxUtil was hiding such errors.   It was a mess, and this eager generation of domain (even for allowDuplicateDomains== false case) was the early mess detector. 

I have another PR fixing this ISPN-10887 so I think you should remove this change to not clobber history.

Reply:
https://github.com/infinispan/infinispan/blob/ca322fdba3b8fef6588b6e62b3092cf4da239511/core/src/main/java/org/infinispan/jmx/CacheManagerJmxRegistration.java#L86 :))))

Reply:
The "early detection" stuff was too unreliable, you would sometimes get an exception here and sometimes a completely different exception when two managers got the same name. IMO it's better to not have any early detection and to treat all `InstanceAlreadyExistsException` exceptions the same.